homeassistant/components/synology_dsm/__init__.py

```diff
@@ -162,7 +162,11 @@ def _async_migrator(entity_entry: entity_registry.RegistryEntry):
     await entity_registry.async_migrate_entries(hass, entry.entry_id, _async_migrator)
 
     # Continue setup
-    await api.async_setup()
+    api = SynoApi(hass, entry)
+    try:
+        await api.async_setup()
+    except Exception as err:  # pylint: disable=broad-except
```

Comment:
Can we distinguish between auth failed and a failure to connect so we don't keep retrying and brute force the box when the password is wrong?

Reply:
Yep we can.
Modified and testing.

Reply:
Almost forgot to push ... 😅 but here we are !